mne_bids/tests/test_mne_bids.py

```diff
@@ -182,21 +189,47 @@ def test_kit():
     # ensure the channels file has no STI 014 channel:
     channels_tsv = make_bids_basename(
         subject=subject_id, session=session_id, task=task, run=run,
-        suffix='channels.tsv', acquisition=acq,
+        suffix='channels.tsv',
         prefix=op.join(output_path, 'sub-01/ses-01/meg'))
     data = _from_tsv(channels_tsv)
     assert 'STI 014' not in data['name']
 
     # ensure the marker file is produced in the right place
     raw_folder = make_bids_basename(
         subject=subject_id, session=session_id, task=task, run=run,
-        acquisition=acq, suffix='%s' % 'meg')
+        suffix='%s' % 'meg')
     marker_fname = make_bids_basename(
         subject=subject_id, session=session_id, task=task, run=run,
-        acquisition=acq, suffix='markers.sqd',
+        suffix='markers.sqd',
         prefix=os.path.join(output_path, 'sub-01/ses-01/meg', raw_folder))
     assert op.exists(marker_fname)
 
+    # test correct naming of marker files
+    raw = mne.io.read_raw_kit(
+        raw_fname, mrk=[hpi_post_fname, hpi_pre_fname], elp=electrode_fname,
+        hsp=headshape_fname)
+    write_raw_bids(raw,
+                   kit_bids_basename.replace('sub-01', 'sub-%s' % subject_id2),
+                   output_path, events_data=events_fname, event_id=event_id,
+                   overwrite=False)
+    cmd = ['bids-validator', output_path]
+    run_subprocess(cmd, shell=shell)
+    # ensure the marker files are renamed correctly
+    raw_folder = make_bids_basename(
+        subject=subject_id2, session=session_id, task=task, run=run,
+        suffix='%s' % 'meg')
+    marker_fname = make_bids_basename(
+        subject=subject_id2, session=session_id, task=task, run=run,
+        suffix='markers.sqd', acquisition='pre',
+        prefix=os.path.join(output_path, 'sub-02/ses-01/meg', raw_folder))
```

Comment:
could these `make_bids_basename` lines be replaced by simple python `str.replace` kind of operations? It makes the code unreadable in my opinion. Something shorter would be nicer.

Reply:
Not always entirely. Because we need to add the `acquisition` parameter which goes *before* task it would be far messier than how we have it here.
I can clear up some but not all I think.